### PR TITLE
upgrade typeorm to 0.3.x, clean up config and build

### PR DESCRIPTION
**Pull request by sgfost**
#### API changes:
https://typeorm.io/changelog#breaking-changes-2

#### DataSource, connection/manager deprecation:
https://typeorm.io/changelog#deprecations

---

resolves virtualcommons/planning#65

### CODE REVIEW

**Comment by sgfost** [comment]
This came dangerously close to making me try out migrating to prisma. Typeorm gets to live another ~~day~~release

**Comment by alee** [comment]
> This came dangerously close to making me try out migrating to prisma. Typeorm gets to live another ~day~release

[prisma](https://www.prisma.io/docs/orm/overview/introduction/what-is-prisma) looks interesting and a simpler way to map JS things to a database than typeorm which is one of those bring-familiar-abstractions-from-another-implementation (Hibernate / JPA / C# Entity Framework) to typescript

**Comment by alee** [comment]
`npm upgrade` fails because of https://github.com/vitejs/vite/issues/15714 - seems we'd need to upgrade to vite 5 (and vue 3?) to fix that. 

pinning `@types/node` to ~18.8 generates more whackamole compilation issues ala https://github.com/microsoft/TypeScript/issues/51567

File: client/src/components/global/Footer.vue [hunk 1]
```diff
@@ -93,7 +93,7 @@ ccarra1@asu.edu
       &copy; 2020-{{ currentYear }}
       <a href="https://www.azregents.edu/">Arizona Board of Regents</a> |
 
-      <a :href="constants.GITHUB_URL">{{ constants.BUILD_ID }}</a>
+      <a :href="constants.GITHUB_URL">{{ constants.RELEASE_VERSION }}</a>
```
Comment (alee) [inline client/src/components/global/Footer.vue:96]: for some reason this isn't being rendered properly on staging deployment, shows up as "unknown". Perhaps  the shared/settings environment isn't being set at runtime?

should probably fully merge Constants into `shared/src/settings` as per its FIXME tag

**Review by alee — COMMENTED** [review]
LGTM though I think we should finish up the shared/src/settings.ts refactor and figure out why staging (and presumably production) isn't rendering `RELEASE_VERSION` properly

File: client/src/components/global/Footer.vue (hunk 1 above)
Comment (sgfost) [reply client/src/components/global/Footer.vue:96]: oh yeah I think it needs access to these at build time for them to be available. I'll try and do this with vite's support for env vars https://vitejs.dev/guide/env-and-mode

**Comment by sgfost** [comment]
shared settings and env vars should be fixed now. I haven't looked into the problems with upgrading yet

**Comment by alee** [comment]
thanks Scott! LGTM, we can deal with dependency upgrades later

**Pull request merged**